addons/cookoff/functions/fnc_detonateAmmunition.sqf

```diff
@@ -1,135 +1,174 @@
+    private _projectile = createVehicle [_ammo, _spawnPos, [], 0, "CAN_COLLIDE"];
 
-    if (toLower _simType == "shotbullet") then {
-        private _sound = selectRandom [QUOTE(PATHTO_R(sounds\light_crack_close.wss)), QUOTE(PATHTO_R(sounds\light_crack_close_filtered.wss)), QUOTE(PATHTO_R(sounds\heavy_crack_close.wss)), QUOTE(PATHTO_R(sounds\heavy_crack_close_filtered.wss))];
-        playSound3D [_sound, objNull, false, (getPosASL _vehicle), 2, 1, 1250];
+    if (_flyAway) then {
+        private _vectorAmmo = [-1 + random 2, -1 + random 2, -0.2 + random 1];
+        private _vectorVelocity = _vectorAmmo vectorMultiply _speed;
+
+        _projectile setVectorDir _vectorVelocity;
+        _projectile setVelocity _vectorVelocity;
+    } else {
+        _projectile setDamage 1;
+    };
+};
+
+switch (_simType) do {
+    case "shotbullet": {
+        private _sound = selectRandom [QPATHTO_R(sounds\light_crack_close.wss), QPATHTO_R(sounds\light_crack_close_filtered.wss), QPATHTO_R(sounds\heavy_crack_close.wss), QPATHTO_R(sounds\heavy_crack_close_filtered.wss)];
+        playSound3D [_sound, objNull, false, getPosASL _object, 2, 1, 1250];
 
         if (random 1 < 0.6) then {
-            [_vehicle, _ammo, _speed, true] call _spawnProjectile;
+            [_object, _ammo, _speed, true] call _fnc_spawnProjectile;
         };
     };
-    if (toLower _simType == "shotshell") then {
-        private _sound = selectRandom [QUOTE(PATHTO_R(sounds\heavy_crack_close.wss)), QUOTE(PATHTO_R(sounds\heavy_crack_close_filtered.wss))];
-        playSound3D [_sound, objNull, false, (getPosASL _vehicle), 2, 1, 1300];
+    case "shotshell": {
+        private _sound = selectRandom [QPATHTO_R(sounds\heavy_crack_close.wss), QPATHTO_R(sounds\heavy_crack_close_filtered.wss)];
+        playSound3D [_sound, objNull, false, getPosASL _object, 2, 1, 1300];
 
         if (random 1 < 0.15) then {
-            [_vehicle, _ammo, _speed, true] call _spawnProjectile;
+            [_object, _ammo, _speed, true] call _fnc_spawnProjectile;
         };
     };
-    if (toLower _simType == "shotgrenade") then {
+    case "shotgrenade": {
         if (random 1 < 0.9) then {
             _speed = 0;
         };
-        [_vehicle, _ammo, _speed, random 1 < 0.5] call _spawnProjectile;
+
+        [_object, _ammo, _speed, random 1 < 0.5] call _fnc_spawnProjectile;
     };
-    if (toLower _simType in ["shotrocket", "shotmissile", "shotsubmunitions"]) then {
+    case "shotrocket";
+    case "shotmissile";
+    case "shotsubmunitions": {
         if (random 1 < 0.1) then {
-            private _sound = selectRandom [QUOTE(PATHTO_R(sounds\cannon_crack_close.wss)), QUOTE(PATHTO_R(sounds\cannon_crack_close_filtered.wss))];
-            playSound3D [_sound, objNull, false, (getPosASL _vehicle), 3, 1, 1600];
+            private _sound = selectRandom [QPATHTO_R(sounds\cannon_crack_close.wss), QPATHTO_R(sounds\cannon_crack_close_filtered.wss)];
```

Comment:
Is it possible to change the sound to be played back to read from cfgSounds?
I would like to make it easier to create a MOD that changes cookoff's sounds.

Reply:
I want to get #5327 merged, which changes the sounds.
It would be pointless doing doing that sort of thing here, as most config entries would change, but I think we can add something to #5327 to make it configurable (would go through `CfgSFX` though).